skyportal/handlers/api/photometry.py

```diff
@@ -674,61 +676,49 @@ def put(self):
                 return self.error(f'Invalid object ID: {oid}')
 
             # make sure no duplicate data are posted
-            existing_photometry = Obj.get_photometry_owned_by_user(
-                oid, self.current_user
-            )
-            ep = [serialize(phot, "ab", "flux") for phot in existing_photometry]
-            df_existing_photometry = pd.DataFrame(ep)
-            if len(df_existing_photometry) > 0:
-                df_existing_photometry["group_ids"] = df_existing_photometry.apply(
-                    lambda row: set([g.id for g in row["groups"]]), axis=1
-                )
-
-                w_oid_duplicate = (df.obj_id == oid) & (
-                    df.hash.isin(df_existing_photometry.hash)
+            duplicated_photometry = (
+                DBSession()
+                .query(Photometry)
+                .filter(
+                    Photometry.obj_id == oid, Photometry.hash.in_(df["hash"].tolist())
                 )
+                .options(joinedload(Photometry.groups))
+                # .all()
+            )
 
-                if any(w_oid_duplicate):
-                    # update photometry.groups if necessary
-                    for row in df.loc[w_oid_duplicate].itertuples():
-                        # select the corresponding row in existing photometry by the pre-computed hash:
-                        w = df_existing_photometry.hash == row.hash
-                        # select the photometry point's group_ids:
-                        row_group_ids = set(
-                            df_existing_photometry.loc[w, "group_ids"].values[0]
+            if duplicated_photometry is not None:
+                existing_hashes = [d.hash for d in duplicated_photometry]
+                # save existing ids:
+                existing_ids.extend([d.id for d in duplicated_photometry])
+
+                # update groups if necessary
+                for duplicate in duplicated_photometry:
+                    duplicate_group_ids = set([g.id for g in duplicate.groups])
+
+                    # posting to new groups?
+                    if len(set(group_ids) - duplicate_group_ids) > 0:
+                        # select old + new groups
+                        group_ids_update = set(group_ids).union(duplicate_group_ids)
+                        groups = (
+                            DBSession()
+                            .query(Group)
+                            .filter(Group.id.in_(group_ids_update))
+                            .all()
```

Comment:
Check for nonexistent groups

Reply:
Added a check for forbidden/non-existent groups